src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -775,6 +776,58 @@ private void VisitObjectCreationExpressionBase(BoundObjectCreationExpressionBase
                         node.ArgsToParamsOpt,
                         _localScopeDepth,
                         _diagnostics);
+
+                    if (node.InitializerExpressionOpt is { })
+                    {
+                        // Object initializers are different than a normal constuctor in that the 
+                        // scope of the receiver is determined by evaluating the inputs to the constructor
+                        // *and* all of the initializers. Another way of thinking about this is that
+                        // every argument in an initializer that can escape to the receiver is 
+                        // effectively an argument to the constructor. That means we need to do
+                        // a second mixing pass here where we consider the receiver escaping 
+                        // back into the ref parameters of the constructor.
+                        //
+                        // At the moment this is only a hypothetical problem. Because the language 
+                        // doesn't support ref field of ref struct mixing like this could not actually
+                        // happen in practice. At the same time we want to error on this now so that 
+                        // in a future when we do have ref field to ref struct this is not a breaking 
+                        // change. Customers can respond to failures like this by putting scoped on
+                        // such parameters in the constructor.
+                        var escapeValues = ArrayBuilder<EscapeValue>.GetInstance();
+                        var escapeTo = GetValEscape(node.InitializerExpressionOpt, _localScopeDepth);
```

Comment:
Comment above says:
> we need to do a second mixing pass here where we consider the receiver escaping back into the ref parameters of the constructor.

Doesn't this escape scope represent the receiver which will possibly escape into the constructor parameters? Since if that receiver *did* have a narrower scope than the parameters, it had to be the object-initializer giving it that scope. That means that this variable should be `escapeFrom`, not `escapeTo`, right?

Reply:
Yep. Good catch.